File: CalculatorPython/MainWindow.py
from PyQt6 import uic
from PyQt6.QtWidgets import QApplication
from PyQt6 import QtCore

Form, Window=uic.loadUiType("mainwindow.ui")
app = QApplication([])
window=Window()
form = Form()
form.setupUi(window)
window.show()


number_first=0
def digit_numbers():
    button = window.sender()
    if form.result_view.text().find('.')>=0 and button.text()=='0':
        s=form.result_view.text()+button.text()
    else:
        number = float(form.result_view.text()+button.text())
        s="{:}".format(int(number)if int(number)==number else float(number))
    form.result_view.setText(s)

form.pushButton_0.clicked.connect(digit_numbers)
form.pushButton_1.clicked.connect(digit_numbers)
form.pushButton_2.clicked.connect(digit_numbers)
form.pushButton_3.clicked.connect(digit_numbers)
form.pushButton_4.clicked.connect(digit_numbers)
form.pushButton_5.clicked.connect(digit_numbers)
form.pushButton_6.clicked.connect(digit_numbers)
form.pushButton_7.clicked.connect(digit_numbers)
form.pushButton_8.clicked.connect(digit_numbers)
form.pushButton_9.clicked.connect(digit_numbers)

def on_pushButton_dot_clicked():
    s=form.result_view.text()
    if s.find('.')<0:
        form.result_view.setText(s+'.')

form.pushButton_dot.clicked.connect(on_pushButton_dot_clicked)

def operation():
    button = window.sender()
    if(button.text()=='+/-'):
        number=float(form.result_view.text())
        number*=-1
        s="{:}".format(int(number)if int(number)==number else float(number))
        form.result_view.setText(s)
    if(button.text()=="AC"):
        number=0
        form.pushButton_plus.setChecked(False)
        form.pushButton_minus.setChecked(False)
        form.pushButton_times.setChecked(False)
        form.pushButton_demiter.setChecked(False)  
        form.result_view.setText(str(number))
    if(button.text()=="%"):
        number=float(form.result_view.text())
        number*=0.01
        s="{:}".format(int(number)if int(number)==number else float(number))
        form.result_view.setText(s)

form.pushButton_plusminus.clicked.connect(operation)
form.pushButton_AC.clicked.connect(operation)
form.pushButton_procent.clicked.connect(operation)

def math_operation():
    global number_first
    button=window.sender()
    button.setChecked(True)
    number_first=float(form.result_view.text())
    form.result_view.setText('0')

def on_pushButton_answer_clicked():
    global number_first
    number_second=float(form.result_view.text())
    res=0

    if form.pushButton_plus.isChecked():
        res = number_first+number_second
        form.pushButton_plus.setChecked(False)     

    if form.pushButton_minus.isChecked():
        res = number_first-number_second
        form.pushButton_minus.setChecked(False)
    
    if form.pushButton_demiter.isChecked():
        if(number_second == 0):
            res = form.result_view.setText('0')
        else:
            res = number_first/number_second
            form.pushButton_demiter.setChecked(False)

    if form.pushButton_times.isChecked():
        res = number_first*number_second
        form.pushButton_times.setChecked(False)

    s="{:}".format(int(res)if int(res)==res else float(res))
    form.result_view.setText(s)
        

form.pushButton_plus.setCheckable(True)
form.pushButton_minus.setCheckable(True)
form.pushButton_times.setCheckable(True)
form.pushButton_demiter.setCheckable(True)

form.pushButton_plus.clicked.connect(math_operation)
form.pushButton_minus.clicked.connect(math_operation)
form.pushButton_times.clicked.connect(math_operation)
form.pushButton_demiter.clicked.connect(math_operation)
form.pushButton_answer.clicked.connect(on_pushButton_answer_clicked)

app.exec()

File: CalculatorCPP/mainwindow.cpp
#include "mainwindow.h"
#include "./ui_mainwindow.h"

double num_first;

MainWindow::MainWindow(QWidget *parent)
    : QMainWindow(parent)
    , ui(new Ui::MainWindow)
{
    ui->setupUi(this);

    connect(ui->pushButton_0,         SIGNAL(clicked()), this, SLOT(digit_numbers()));
    connect(ui->pushButton_1,         SIGNAL(clicked()), this, SLOT(digit_numbers()));
    connect(ui->pushButton_2,         SIGNAL(clicked()), this, SLOT(digit_numbers()));
    connect(ui->pushButton_3,         SIGNAL(clicked()), this, SLOT(digit_numbers()));
    connect(ui->pushButton_4,         SIGNAL(clicked()), this, SLOT(digit_numbers()));
    connect(ui->pushButton_5,         SIGNAL(clicked()), this, SLOT(digit_numbers()));
    connect(ui->pushButton_6,         SIGNAL(clicked()), this, SLOT(digit_numbers()));
    connect(ui->pushButton_7,         SIGNAL(clicked()), this, SLOT(digit_numbers()));
    connect(ui->pushButton_8,         SIGNAL(clicked()), this, SLOT(digit_numbers()));
    connect(ui->pushButton_9,         SIGNAL(clicked()), this, SLOT(digit_numbers()));
    connect(ui->pushButton_plusminus, SIGNAL(clicked()), this, SLOT(operation()));
    connect(ui->pushButton_proccent,  SIGNAL(clicked()), this, SLOT(operation()));
    connect(ui->pushButton_ac,        SIGNAL(clicked()), this, SLOT(operation()));
    connect(ui->pushButton_demiter,   SIGNAL(clicked()), this, SLOT(math_operation()));
    connect(ui->pushButton_plus,      SIGNAL(clicked()), this, SLOT(math_operation()));
    connect(ui->pushButton_minus,     SIGNAL(clicked()), this, SLOT(math_operation()));
    connect(ui->pushButton_times,     SIGNAL(clicked()), this, SLOT(math_operation()));

    ui->pushButton_demiter  ->  setCheckable(true);
    ui->pushButton_plus     ->  setCheckable(true);
    ui->pushButton_minus    ->  setCheckable(true);
    ui->pushButton_times    ->  setCheckable(true);
}

MainWindow::~MainWindow()
{
    delete ui;
}

void MainWindow :: digit_numbers(){
    QPushButton *button = (QPushButton *)sender();

    double all_numbers;
    QString new_label;


    all_numbers = (ui->result_show->text()+button->text()).toDouble();
    new_label = QString::number(all_numbers, 'g', 10);

    if(ui->result_show->text().contains(".") && button->text() == "0") new_label = ui->result_show->text()+button->text();

    ui -> result_show->setText(new_label);

}

void MainWindow::on_pushButton_dot_clicked()
{
    if(!(ui->result_show->text().contains('.'))) ui -> result_show->setText(ui->result_show->text()+".");
}

void MainWindow :: operation(){
    QPushButton *button = (QPushButton *)sender();

    double all_numbers;
    QString new_label;

    if(button-> text()=="+/-"){
        all_numbers = (ui->result_show->text()).toDouble();
        all_numbers = all_numbers*(-1);
        new_label = QString::number(all_numbers, 'g', 10);

        ui -> result_show->setText(new_label);
    }
    if(button-> text()=="%"){
        all_numbers = (ui->result_show->text()).toDouble();
        all_numbers = all_numbers * 0.01;
        new_label = QString::number(all_numbers, 'g', 10);

        ui -> result_show->setText(new_label);
    }
    if(button-> text()=="AC"){
        all_numbers = 0;
        new_label = QString::number(all_numbers, 'g', 10);

        ui -> result_show->setText(new_label);
        ui -> pushButton_plus->setChecked(false);
        ui -> pushButton_minus->setChecked(false);
        ui -> pushButton_times->setChecked(false);
        ui -> pushButton_demiter->setChecked(false);
    }
}

void MainWindow :: math_operation(){
    QPushButton *button = (QPushButton *)sender();

    num_first = ui->result_show->text().toDouble();
    ui->result_show->setText("0");

    button->setChecked(true);
}


void MainWindow::on_pushButton_answer_clicked()
{

    double label_number;
    double num_second;
    QString new_label;

    num_second = ui->result_show->text().toDouble();


    if(ui->pushButton_plus->isChecked()){
        label_number = num_first + num_second;
        new_label = QString::number(label_number, 'g', 10);

        ui->result_show->setText(new_label);
        ui->pushButton_plus->setChecked(false);
    }else if(ui->pushButton_minus->isChecked()){
        label_number = num_first - num_second;
        new_label = QString::number(label_number, 'g', 10);

        ui->result_show->setText(new_label);
        ui->pushButton_minus->setChecked(false);
    } else if(ui->pushButton_demiter->isChecked()){

        if(num_second == 0) ui->result_show->setText("3rr0r");
        else{
        label_number = num_first / num_second;
            new_label = QString::number(label_number, 'g', 10);

            ui->result_show->setText(new_label);
            ui->pushButton_demiter->setChecked(false);
        }
    }else if(ui->pushButton_times->isChecked()){
        label_number = num_first * num_second;
        new_label = QString::number(label_number, 'g', 10);

        ui->result_show->setText(new_label);
        ui->pushButton_times->setChecked(false);
    }
}

